python/cudf/cudf/core/column/decimal.py

```diff
@@ -229,6 +229,18 @@ def fillna(
         )
         return self._copy_type_metadata(result)
 
+    def serialize(self) -> Tuple[dict, list]:
+        header, frames = super().serialize()
+        header["dtype"] = self.dtype.serialize()
+        header["size"] = self.size
```

Comment:
Does the general `Column.serialize()` not handle the size already? If it doesn't why do we need to handle it here? Similarly, how does it handle dtype that doesn't work as expected here?

Reply:
It does not currently handle `size`. This PR should make it so that it does, if it's part of the header (there's a new `.get` call included in these changes). 

As for dtypes, for most numeric types, `.str` is all the information it needs to unpack the type - eventually this falls into `build_column` and then there's a `pd.api.types.pandas_dtype(dtype)` that turns it back into a numpy type from a string. 

The problem is that since a decimaldtype isn't quite as simple as a raw numeric type, and there's no numpy equivalent, there's no way to write a `cudf.Decimal64Dtype.str` that `pd.api.types.pandas_dtype` can unpack and somehow return the right cudf object.